draft-ietf-tls-esni.md

```diff
@@ -826,11 +826,15 @@ When the server sends a HelloRetryRequest, the client determines if ECH was
 accepted by checking the message for an "encrypted_client_hello" extension with
 an 8-byte payload equal to `hrr_accept_confirmation` as defined in
 {{backend-server}}. If found, the client presumes acceptance and handles the
-HelloRetryRequest using ClientHelloInner. Otherwise, it presumes rejection and
-handles the HelloRetryRequest using ClientHelloOuter. Note that the
-client-facing server does not send "encrypted_client_hello" in case of
+HelloRetryRequest using ClientHelloInner. Otherwise, if the extension is missing
+or its payload is not equal to `hrr_accept_confirmation`, the client presumes
+rejection and handles the HelloRetryRequest using ClientHelloOuter. Note that
```

Comment:
What if the payload is present but malformed, e.g., 1 byte long? Would we want the client to abort then? (Failing early and loudly)

Reply:
A 1-byte *can't* equal `hrr_accept_confirmation`, so the client would presume rejection.

Reply:
Right, but peers typically abort when values are malformed or otherwise incorrect. Presuming rejection in this case seems to break with that pattern.

Reply:
Offline we agreed that the client should abort if the payload is malformed.